### PR TITLE
Milestone 2: Components

### PR DESCRIPTION
**Pull request by Ceci007**
## Milestone 2: Components
In this milestone I:
- Created a components directory in the src folder.
- Created four files in there, each one representing one React component.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import Display from './Display';
+import ButtonPannel from './ButtonPannel';
+
+const App = () => (
+  <div className="App">
+    <Display />
+    <ButtonPannel />
+  </div>
```
Comment (bellom) [inline src/components/App.js:9]: We need to use [react fragment](https://reactjs.org/docs/fragments.html) in here;

image


We don't want to create an extra element, that's why you need to use fragment, so we are able to render without having to create an extra node wrapping `Display` and `ButtonPanel`.

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import '../index.css';
+import PropTypes from 'prop-types';
+
+const Button = ({ name, color, size }) => (
+  <button type="button" className={`${color} ${size}`}>{ name }</button>
+);
```
Comment (bellom) [inline src/components/Button.js:7]: According to the pathwright, I think you just need to pass `name` right?
image

Can we know the reason why you added `color` and `size`?

File: src/components/ButtonPannel.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import '../index.css';
+import Button from './Button';
+
+function ButtonPannel() {
```
Comment (bellom) [inline src/components/ButtonPannel.js:5]: ### [Optional]
I think you can simplify this logic like this 👇 and I as said I don't think we need `size` and `color`.

You can use [objects.key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)  and `map` to loop through the objects. `objects.keys` for the outer loop and `map` for the inner loop.

Example; 
```
const GroupButtons = {
  Group 1: ['AC', '+/-', '%', '÷'],
  Group 2: ['7', '8', '9', 'X'],
   ............
}
```

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Ceci007  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/components/Button.js (hunk 2 above)
Comment (Ceci007) [reply src/components/Button.js:7]: I was thinking that in the future I could add some dynamic styling, but you are right that is not an instruction for this milestone, maybe I can save that for future milestones. sorry for that.

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: :clap: 
Hi @Ceci007 :hand: 
Awesome work on this milestone and applying the requested changes as well.
Good luck with the next one.

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**